nativelib/src/main/scala/scala/scalanative/process/WindowsUtils.scala

```diff
@@ -0,0 +1,48 @@
+package scala.scalanative.process
+
+// Required only for cross-compilation with Scala 2
+import scala.language.existentials
+
+object WindowsUtils {
+
+  // https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-commandlinetoargvw#remarks
+  def argvToCommand(argv: java.util.Iterator[String]): String = {
+    val sb = new java.lang.StringBuilder
+    val sbBackslash = new java.lang.StringBuilder
+    def appendBackslashes(double: Boolean): Unit = {
+      val cnt = sbBackslash.length()
+      if (cnt != 0) {
+        sb.append(sbBackslash, 0, cnt)
+        if (double) sb.append(sbBackslash, 0, cnt)
+        sbBackslash.setLength(0)
+      }
+    }
+
+    argv.forEachRemaining { arg =>
+      if (sb.length() != 0) sb.append(' ')
+      if (arg.isEmpty || arg.exists(c => c.isWhitespace || c == '"')) {
+        sb.append('"')
+        arg.foreach { c =>
+          if (c == '\\') sbBackslash.append(c)
+          else {
+            val isQuote = c == '"'
+            appendBackslashes(double = isQuote) // double them before a quote
+            if (isQuote) sb.append('\\') // escape the quote
+            sb.append(c)
+          }
+        }
+        appendBackslashes(double = true) // if trailing
+        sb.append('"')
+      } else sb.append(arg)
+    }
+
+    sb.toString
+  }
+
+  def argvToCommand(argv: scala.collection.Iterator[String]): String =
```

Comment:
Is this only used in the test? We are trying to keep `javalib` free of Scala collections.

Reply:
yes, in the test. i could move it there.

yet, when you say `javalib`, does `nativelib`/`scalanative` count? :)

Reply:
done

Reply:
We are just trying to minimize the dependency of `scalalib` from `javalib`. Scala.js has eliminated the dependency entirely but they use a code cleaner so we are trying to potentially support that same approach. It was unclear if this was really part of `javalib` and just in the `scalanative` package. I didn't check but it seemed to be.

Reply:
Use of Scala in the tests is fine. Only in `javalib` code and the code it calls (hopefully).